### DISCUSSION TITLE
Timezone when running from source

### DISCUSSION THREAD

**Discussion by spikeh** [philosowaffle/peloton-to-garmin#454] [Q&A]
Hi, I'm running peloton-to-garmin from source. All uploads are in an incorrect timezone (UTC-6), and I ensured that my Garmin Connect account, watch and Peloton bike are all configured to the correct timezone (UTC-8).

<img width="551" alt="image" src="https://user-images.githubusercontent.com/1028992/224520576-d7b1b092-918e-41fb-be17-29a63e250b72.png">

I double checked that the timezone on my computer is also correct:
```
$ timedatectl
               Local time: Sat 2023-03-11 18:18:54 PST
           Universal time: Sun 2023-03-12 02:18:54 UTC
                 RTC time: Sat 2023-03-11 18:18:54
                Time zone: America/Los_Angeles (PST, -0800)
System clock synchronized: yes
              NTP service: active
          RTC in local TZ: no
```

I saw a different discussion that said to pass the correct timezone via an env var to Docker. What is the correct way to do this when running from source?

**Comment by philosowaffle** [comment]
When running from source, it should be using the local computers timezone. So not sure why its not working in this case.

**Comment by spikeh** [comment]
@philosowaffle Could you please provide some code pointers as to where the timezone is fetched and timestamp calculated? I can take a look.

**Reply by philosowaffle** [reply]
[This is what the raw StartTime](https://github.com/philosowaffle/peloton-to-garmin/blob/master/src/UnitTests/Data/p2g_workouts/cycling_workout.json#L16) coming from Peloton looks like. [This is where P2G calculates the start time](https://github.com/philosowaffle/peloton-to-garmin/blob/master/src/Conversion/IConverter.cs#L205) that gets sent to Garmin.  There's also [this bit which could be the culprit](https://github.com/philosowaffle/peloton-to-garmin/blob/master/src/Conversion/FitConverter.cs#L195) and also gets sent to Garmin.
